cfspopcon/key_checker.py

```diff
@@ -0,0 +1,195 @@
+        issue_detected = False
+
+        issue_detected = self.fix_unit_dictionary(dry_run) or issue_detected
+        issue_detected = self.fix_glossary(dry_run) or issue_detected
+
+        return issue_detected
+
+    def read_physics_glossary(self) -> tuple[dict[str, list[str]], set[str]]:
+        """Read the physics glossary file."""
+        with as_file(files("cfspopcon").parents[0] / "docs" / "doc_sources" / "physics_glossary.rst") as filepath:  # type:ignore[attr-defined]
+            glossary_text = Path(filepath).read_text().splitlines()
+
+        # Find the length of the header
+        header_line = 0
+        for i, line in enumerate(glossary_text):
+            if ":sorted:" in line:
+                header_line = i
+                break
+        if header_line == 0:
+            raise RuntimeError("Header not found")
+        self._glossary_header = glossary_text[: header_line + 1]
+
+        # Define patterns for identifying blank lines, keys and descriptions.
+        pattern_for_key = re.compile(r"^\s{2}[^\s]")
+        pattern_for_description = re.compile(r"\s{4}[^\s]")
+        pattern_for_blank = re.compile(r"^\s*$")
+
+        # Store the results in a dictionary
+        glossary: dict[str, list[str]] = dict()
+
+        # Iterate over the glossary, ignoring the header lines
+        keys: list[str] = []
+        description: list[str] = []
+        for line in glossary_text[header_line + 1 :]:
+            line_is_blank = pattern_for_blank.match(line)
+            line_is_key = pattern_for_key.match(line)
+            line_is_description = pattern_for_description.match(line)
+
+            if not (line_is_key or line_is_description):
+                # Blank lines indicate new entry
+                assert line_is_blank, f"Line didn't match format for key or description, and wasn't blank. Line was: '{line}'"
+
+                if keys or description:
+                    if len(keys) > 1:
+                        raise RuntimeError(f"Multiple keys for entry: ({keys}). This must be fixed before proceeding.")
+                    glossary[keys[0]] = description
+
+                keys = []
+                description = []
+
+            if line_is_key:
+                keys.append(line.strip())
+
+            if line_is_description:
+                description.append(line.strip())
+
+        if keys or description:
+            if len(keys) > 1:
+                raise RuntimeError(f"Multiple keys for entry: ({keys}). This must be fixed before proceeding.")
+            glossary[keys[0]] = description
+
+        return glossary, set(glossary.keys())
+
+    def read_algorithm_keys(self) -> set[str]:
+        """Read the algorithm keys."""
+        algorithm_keys = []
+
+        for alg in Algorithm.instances.values():
+            for key in alg.input_keys + alg.return_keys:
+                if key not in algorithm_keys:
+                    algorithm_keys.append(key)
```

Comment:
What do you think about directly making it a set and using `update` ? 
```suggestion
        algorithm_keys: set[str] = {}

        for alg in Algorithm.instances.values():
            algorithm_keys.update(alg.input_keys)
            algorithm_keys.update(alg.return_keys)
```